### DISCUSSION TITLE
NFTs & PINATA

### DISCUSSION THREAD

**Discussion by oscarmairey** [smartcontractkit/full-blockchain-solidity-course-py#397] [Q&A] [answered]
Hello, I have kind of a strange error when uploading with Pinata.
![image](https://user-images.githubusercontent.com/92814105/142476466-3bd46b40-58ef-497b-a6f6-b57b967846e9.png)

Here is my code which i copy pasted fom Patrick's code: 
![image](https://user-images.githubusercontent.com/92814105/142476536-16e5993e-fe1a-4f14-b370-0adfadb9ac9d.png)

I litteraly tried everything...

**Comment by cromewar** [comment, marked as answer]
Hello @Cesarioo, make sure to not add the pinata API keys on your `.env` file using "", they should look like this:

![image](https://user-images.githubusercontent.com/20199180/142479403-6eb19802-87fa-43dc-a399-cd9eff340aaa.png)

Remember sometimes brownie does no recognize recently added env variables, so make sure you have `dotenv: env` on `brownie-config.yaml`, then restart vscode, delete the build folder and compile again.

Best regards, Cromewar

**Reply by oscarmairey** [reply]
This does not work. 
![image](https://user-images.githubusercontent.com/92814105/142489211-1d8766cc-4600-46f6-b655-6263322ba967.png)
here are my .env pinata apis
and I tried to print the secret PINATA and here is the outcome 
![image](https://user-images.githubusercontent.com/92814105/142489604-25039572-a8eb-40d2-bf13-7bafd8e3f52e.png)
So everything is normal on the .env side

**Reply by cromewar** [reply]
I'll insist the problem is on the .env variables, I managed to reproduce the problem:

![image](https://user-images.githubusercontent.com/20199180/142498628-1808759a-ed0b-4088-bc51-864c4a09d13c.png)

Importing the env variables:

![image](https://user-images.githubusercontent.com/20199180/142498708-cee278b3-e87c-4da1-b513-c228514fdb34.png)

Everything works fine ;)

![image](https://user-images.githubusercontent.com/20199180/142498746-6ed1826e-d926-4e16-88dd-cf162b003b61.png)

Give it a try.

**Reply by oscarmairey** [reply]
Thank you for your answers.
Finally it was the .env
Since I was using windows env variables I decided to create new ones on VSCode and now it works perfectly. I think Windows does not store env variables as strings.

I have a question however: why use pinata and not IPFS directly ?

**Reply by cromewar** [reply]
Hello @Cesarioo I'm glad you solved your problem. 

I have a question however: why use pinata and not IPFS directly ?

I think it's a matter of making things easier, if you upload your image to ipfs the only way to keep accessing it is for you to have a node permanently running on your pc,  suing pinata or moralis allows you to use external servers for free.

**Reply by oscarmairey** [reply]
That's what everyone has been saying
However when I shut down daemon the link is still accessible
